### ISSUE TITLE
Add a way for modifications to extend existing eras

### ISSUE THREAD

**Issue by czyzby** [wesnoth/wesnoth#6424]
Extending the default era is a common first step when modders add their own faction or units. `[modify_unit_type]` WML tag with `set_advances_to` and `add_advancement` properties provides a clean way of extending the default units with new advancements in eras, modifications, and reusable resources. However, modifying an existing era like *Default* without downright copying it and creating a new one ranges from tedious to not possible.

The current approach of creating a custom era for a new faction leads to a lot of duplication and incompatibilities. `"Default +"` alone gives 8 matches in the 1.14 add-on server, and there are other eras and modifications that are very similar. The thing is, all of these add-ons are incompatible with each other, even though the majority simply come down to adding one or several new factions meant to fit the mainline eras. By allowing to extend an era via a modification, all of these factions could potentially be compatible with each other - without resorting to other add-ons with the sole purpose of content aggregation. In fact, even _Dunefolk_ could be made available through a mainline modification instead of providing 2 sets of very similar eras.

I propose to add WML tags or Lua functions that allow to do the following:

* Add a new faction to an existing era.
* Remove a faction from an existing era.
* Add a new recruit to a selected faction of an existing era.
* Remove a recruit from a selected faction of an existing era.
* Add a new leader to a selected faction of an existing era.
* Remove a leader from a selected faction of an existing era.

Note that it must be possible to make the features optional and limited to the scope of the add-on, i.e. the existing eras should not be permanently modified by installing an add-on. These features should be possible to use in the scope of a `[resource]`, so that they can be easily included in an era, campaign, or modification.

This will probably lead to less overall eras available in the add-on server, but:

1. The leftover eras will be truly standalone and provide a new game experience.
2. Smaller add-ons with one or few new factions will be compatible with each other and might see more use.
3. Features that can already be added with some effort through WML/Lua like the new recruits will be standardized.

**Comment by knyghtmare** [comment]
Maybe something like these?
```ini
[modify_faction]
    id="drakes_default"
    add_leader="Saurian Spearthrower"
    add_random_leader="Saurian Spearthrower"
[/modify_faction]
```

```ini
[modify_faction]
    id="drakes_aoh"
    add_leader="Saurian Warden","Saurian Seer","Saurian Prophet"
    add_random_leader="Saurian Warden","Saurian Seer","Saurian Prophet"
    add_recruit="Saurian Spearthrower"
[/modify_faction]
```

**Comment by czyzby** [comment]
Yeah, that's what I had in mind. Could also use another block for modifying eras.

```ini
[modify_era]
    id="default"
    [multiplayer_side]
        # ...
    [/multiplayer_side]
    [remove_multiplayer_side]
        id="drakes_default"
    [/remove_multiplayer_side]
[/modify_era]
```
